dms/controllers/portal.py

```diff
@@ -170,6 +138,104 @@ def portal_my_dms_directory(
         }
         return request.render("dms.portal_my_dms", values)
 
+    def _get_files(self, access_token, dms_directory_id, search, search_in, sort_br):
+        """
+        Get files from dms_directory_id
+
+        :param Optional[str] access_token: access_token
+        :param int dms_directory_id: dms_directory_id
+        :param Optional[str] search: search
+        :param Optional[str] search_in: search_in
+        :param str sort_br: sort_br
+
+        :return: dms_file_items
+        :rtype: odoo.model.dms_file
+        """
+        allowed_directory_from_access_group = (
+            request.env["dms.access.group"]
+            .search([("users", "in", request.env.uid)])
+            .directory_ids
+        )
+        has_all_access = request.env["res.users"]._has_all_access(request.env.uid)
+        if (
+            dms_directory_id not in allowed_directory_from_access_group.ids
+            and not has_all_access
+        ):
```

Comment:
I believe that this is not correct in all cases and should look something similar to
`res = self._dms_check_access("dms.directory", dms_directory_id, access_token)`